firesql/sql/sql_fire_query.py

```diff
@@ -123,15 +123,18 @@ def _init_query_refs(self, query: SQL_BinaryExpression, fireQueries={}):
   def _init_column_names(self):
     self.columnNameMap = {}
     colNames = [c.column for c in self.columns]
+    colAliases = [c.alias for c in self.columns]  # new
     for ci in range(len(colNames)):
       tableName = self.aliases[ self.columns[ci].table ] if self.columns[ci].table else self.defaultPart
       if tableName not in self.columnNameMap:
         self.columnNameMap[ tableName ] = {}
-      self.columnNameMap[ tableName ][ colNames[ci] ] = self.columns[ci].column
+      # self.columnNameMap[ tableName ][ colNames[ci] ] = self.columns[ci].column
+      self.columnNameMap[ tableName ][ colNames[ci] ] = colAliases[ci] if colAliases[ci] is not None else self.columns[ci].column # new
       for cj in range(ci+1, len(colNames)):
         if colNames[ci] == colNames[cj]:
           # rename ambiguous column by renaming to table_column
-          self.columnNameMap[ tableName ][ colNames[ci] ] = '_'.join( [self.columns[ci].table, self.columns[ci].column] )
+          # self.columnNameMap[ tableName ][ colNames[ci] ] = '_'.join( [self.columns[ci].table, self.columns[ci].column] )
+          self.columnNameMap[ tableName ][ colNames[ci] ] = '_'.join( [tableName, self.columns[ci].column] )  # new
```

Comment:
If a "table" name has been aliased, previously this would result in prepending the alias to the ambiguous column name.  Instead, now,. it prepends the "table" name that was aliased, not the alias.  I think this minimizes ambiguity in interpretation.  What are you thoughts?

Reply:
Making perfect sense, I like it.